### DISCUSSION TITLE
Does SHAP work for multimodal model that has continuous, categorical, text and image features? Any workable example?

### DISCUSSION THREAD

**Discussion by wjlgatech** [shap/shap#2375] [Q&A]
I am building a Tensorflow based multimodal classifier that consumes tabular data with cnt, cat, txt, img columns. How to explain such model with Shap? Any advice and workable examples would be much appreciated.

I include some notebooks here:
https://github.com/wjlgatech/tf-multimodal

**Comment by jainraj** [comment]
You could use Gradient Explainer to get SHAP values at the level of each dimension. You could then add the SHAP values across the dimensions which make up a logical feature e.g., add up embedding layer SHAP values to get SHAP for the word (very common trick - [TDS blog by Conor Sullivan](https://towardsdatascience.com/shap-for-categorical-features-7c63e6a554ea), [Fiddler.AI blog](https://www.fiddler.ai/blog/should-you-explain-your-predictions-with-shap-or-ig))

**Comment by Matt23-star** [comment]
Hey, have you found a solution for this problem?
